modules/persistence/src/services/connector/index.ts

```diff
@@ -0,0 +1,20 @@
+import * as mongodb from 'mongodb';
+import c from 'config';
+
+const db = async () => {
+  const atlasURL = `mongodb+srv://${process.env.DB_USERNAME}:${process.env.DB_PASSWORD}@${c.get(
+    'dbHost'
+  )}/?retryWrites=true&w=majority`;
+  const client = new mongodb.MongoClient(atlasURL);
+  await client.connect();
+  return client.db(process.env.DB_NAME);
+};
+
+// all docs should be inserted with the buildId for the run.
+export const insert = async (docs, collection, buildId) => {
+  const insertSession = await db();
```

Comment:
similar here for error catching

Reply:
i see we are doing all error handling at caller level. kinda leaving it at the hands of caller, would be safer to catch and throw errors directly from external service calls